From f25c4ec1e4d754275873a7d77c05b79ef6c98148 Mon Sep 17 00:00:00 2001
From: Jiucheng Xu <Jiucheng.Xu@freescale.com>
Date: Thu, 24 Oct 2013 13:17:31 +0800
Subject: [PATCH 1263/1336] Add RSA keygen request for SKMM

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

Signed-off-by: Jiucheng Xu <Jiucheng.Xu@freescale.com>
Change-Id: I10e8467953cf8524a2848f4b474952ea54ed7868
Reviewed-on: http://git.am.freescale.net:8181/6049
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Thomas Trefny <Tom.Trefny@freescale.com>
Reviewed-on: http://git.am.freescale.net:8181/10198
Reviewed-by: Mingkai Hu <Mingkai.Hu@freescale.com>
Reviewed-by: Jose Rivera <German.Rivera@freescale.com>
---
 include/linux/crypto.h | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/include/linux/crypto.h b/include/linux/crypto.h
index 9f10fb5..ffc7468 100644
--- a/include/linux/crypto.h
+++ b/include/linux/crypto.h
@@ -205,6 +205,7 @@ struct aead_request {
 };
 
 enum pkc_req_type {
+	RSA_KEYGEN,
 	RSA_PUB,
 	RSA_PRIV_FORM1,
 	RSA_PRIV_FORM2,
@@ -296,6 +297,27 @@ struct rsa_priv_frm3_req_s {
 	uint32_t c_len;
 };
 
+/*
+ * RSA keygen request
+ */
+
+struct rsa_keygen_req_s {
+	uint8_t *p;
+	uint8_t *q;
+	uint8_t *dp;
+	uint8_t *dq;
+	uint8_t *d;
+	uint8_t *n;
+	uint8_t *c;
+	uint32_t p_len;
+	uint32_t q_len;
+	uint32_t dp_len;
+	uint32_t dq_len;
+	uint32_t d_len;
+	uint32_t n_len;
+	uint32_t c_len;
+};
+
 /* DLC and ECC Keygen request
  @len_L - size of the field
  @len_N - size of the group
@@ -477,6 +499,7 @@ struct pkc_request {
 	enum pkc_req_type type;
 	enum curve_t curve_type;
 	union {
+		struct rsa_keygen_req_s rsa_keygen;
 		struct rsa_pub_req_s rsa_pub_req;
 		struct rsa_priv_frm1_req_s rsa_priv_f1;
 		struct rsa_priv_frm2_req_s rsa_priv_f2;
-- 
2.0.2

